src/data_structures.rs

```diff
@@ -202,46 +201,41 @@ impl<C: PCCommitment> LabeledCommitment<C> {
     }
 
     /// Retrieve the commitment from `self`.
-    pub fn commitment(&self) -> &C {
+    pub fn commitment(&self) -> &G {
         &self.commitment
     }
 
-    /// Retrieve the degree bound in `self`.
-    pub fn degree_bound(&self) -> Option<usize> {
-        self.degree_bound
-    }
-
     /// Randomize commitment for test purpose
-    pub fn randomize(&mut self) {
-        self.commitment.randomize();
+    // #[cfg(test)]
+    pub fn ut_randomize(&mut self) {
+        self.commitment *= &G::ScalarField::rand(&mut rand::thread_rng())
     }
 }
 
-impl<C: PCCommitment> SemanticallyValid for LabeledCommitment<C> {
+impl<G: Group> SemanticallyValid for LabeledCommitment<G> {
     fn is_valid(&self) -> bool {
         self.commitment.is_valid()
     }
 }
 
-impl<C: PCCommitment> ToBytes for LabeledCommitment<C> {
+impl<G: Group> ToBytes for LabeledCommitment<G> {
     #[inline]
-    fn write<W: Write>(&self, writer: W) -> IoResult<()> {
-        self.commitment
-            .serialize_without_metadata(writer)
+    fn write<W: Write>(&self, mut writer: W) -> IoResult<()> {
+        CanonicalSerialize::serialize(&self.commitment, &mut writer)
```

Comment:
There was a reason for the usage of 'serialize_without_metadata' in place of 'serialize'. Whenever we absorb this stuff for hashing we do not want to save info about length of vectors and so on, and 'serialize_without_metadata' ignores them, while 'serialize' doesn't. In this case it doesn't matter, as commitment is a Group implementer while before it was a PCCommitment implementer (that could've been a composite data structure) but, in general, since we are refactoring, I would follow the approach of replacing all ToBytes/FromBytes implementation with CanonicalSerialize/CanonicalDeserialize and call 'serialize_without_metadata' function in place of to_bytes! macro invocations

Reply:
Actually same as above, can do it, this is just bit other step of refactoring which should be done across all repos

Reply:
See above, we can start doing it here